bom/public/pom.xml

```diff
@@ -0,0 +1,254 @@
+                <artifactId>hibernate-search-backend-elasticsearch</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-backend-elasticsearch-aws</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-backend-lucene</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-mapper-pojo-base</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-mapper-pojo-standalone</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-mapper-orm</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-v5migrationhelper-engine</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-v5migrationhelper-orm</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-mapper-orm-batch-jsr352-core</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-mapper-orm-batch-jsr352-jberet</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate.search</groupId>
+                <artifactId>hibernate-search-mapper-orm-coordination-outbox-polling</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+
+            <!-- Import Hibernate ORM BOM -->
+            <dependency>
+                <groupId>org.hibernate.orm</groupId>
+                <artifactId>hibernate-platform</artifactId>
+                <version>${version.org.hibernate.orm}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
```

Comment:
I'm late to the party but: the best practice for a low level library BOM is to only include in the BOM the artifacts you produce and not the artifacts that you depend on.
The case of Quarkus is a bit different because we want to provide a cohesive platform.

I strongly suggest that, In the case of Hibernate Search, the BOM should only contain the Hibernate Search artifacts and that's it. Also I don't think you should use the Quarkus plugin to generate a BOM.

Last but not least, check  your BOM with `mvn help:effective-pom`  at the end to make sure it's lean and that it only contains the Hibernate Search artifacts.

I don't remember all the research we did when we worked on that but if you need further details, @aloubyansky might be able to help.

Reply:
Okay... I think it makes sense. At the very least it's a more conservative approach.

So, my description on the ticket was wrong. Sorry about that @marko-bekhta .

In practice I suppose this means moving all non-Hibernate-Search artifacts (even the Hibernate ORM BOM) out of the `bom/public/pom.xml` into `bom/build/pom.xml`, and changing a bit the documentation since using the Hibernate Search BOM won't solve any of the problems mentioned there anymore.

Does it make sense @marko-bekhta?

Reply:
Hey, no worries 😃 
Yes, it makes sense. As for the docs, I assume we could just "solve it" by importing two BOMs ORM/Search ones instead. I'll test if that will work 😃 

> Also I don't think you should use the Quarkus plugin to generate a BOM.

As for the plugin, we had some artifacts to exclude so that was a useful tool. With ORM BOM gone, we wouldn't need to exclude anything. From what I've seen the plugin creates a resulting pom similar to what `help:effective-pom` produces, resolving all the properties and creating a single XML. Otherwise, we'd have a BOM referencing the root Search pom. I liked the "clean" one more 😃, but if the one with a root parent and unresolved properties is OK I think we can remove the plugin entirely.

Reply:
> Also I don't think you should use the Quarkus plugin to generate a BOM.

Flattening is a good idea. Typically, people use https://www.mojohaus.org/flatten-maven-plugin for this.
You could also use the Quarkus platform plugin, if you like, which was specialized for Quarkus platform BOM flattening. In case it's relevant, it allows configuring a few options such as `excludeArtifactKeys` (exclude some constraints from the original BOM, possibly inherited from an external POM) and `filterInvalidConstraints`, which, if enabled, will validate each constraint in the BOM and exclude those that don't seem to exist.